File: app/src/main/java/com/snowdango/numac/AppListView/AppListCreate.java
package com.snowdango.numac.AppListView;

import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageInfo;
import android.content.pm.PackageManager;

import com.snowdango.numac.ListFormat.AppListFormat;

import java.util.ArrayList;
import java.util.List;
import java.util.Random;

public class AppListCreate{

    public List<AppListFormat> appRead(Context context){

        List<AppListFormat> data = new ArrayList<>();

        PackageManager pm = context.getApplicationContext().getPackageManager();
        List<PackageInfo> pkgInfoList = pm.getInstalledPackages(0);

        Intent i = new Intent(Intent.ACTION_MAIN, null);
        i.addCategory(Intent.CATEGORY_LAUNCHER);

        for (PackageInfo pckInfo : pkgInfoList) {
            if (pm.getLaunchIntentForPackage(pckInfo.packageName) != null) {
                AppListFormat appInfo = new AppListFormat();
                appInfo.setAppIcon(pckInfo.applicationInfo.loadIcon(pm));
                appInfo.setAppName((String) pckInfo.applicationInfo.loadLabel(pm));
                appInfo.setAppPackageName(pckInfo.packageName);
                appInfo.setAppClassName(pm.getLaunchIntentForPackage(pckInfo.packageName).getComponent().getClassName() + "");
                appInfo.setAppCommand(randomCommand());
                data.add(appInfo);
            }
        }

        return data;
    }


    public String randomCommand(){
        Random r = new Random();
        int num = 0;
        String appCommand = "";
        for(int i = 0; i < 4; i++){
            num = r.nextInt(10);
            appCommand += num;
        }
        if (appCommand.equals("0000")) {
            appCommand = randomCommand();
        }
        return  appCommand;
    }
}

File: app/src/main/java/com/snowdango/numac/AppListView/AppListCreate.kt
package com.snowdango.numac.AppListView

import android.content.Context
import android.content.Intent
import com.snowdango.numac.ListFormat.AppListFormat
import java.util.*

class AppListCreate {
    fun appRead(context: Context): List<AppListFormat> {
        val data: MutableList<AppListFormat> = ArrayList()
        val pm = context.applicationContext.packageManager
        val pkgInfoList = pm.getInstalledPackages(0)
        val i = Intent(Intent.ACTION_MAIN, null)
        i.addCategory(Intent.CATEGORY_LAUNCHER)
        for (pckInfo in pkgInfoList) {
            if (pm.getLaunchIntentForPackage(pckInfo.packageName) != null) {
                val appInfo = AppListFormat()
                appInfo.appIcon = pckInfo.applicationInfo.loadIcon(pm)
                appInfo.appName = pckInfo.applicationInfo.loadLabel(pm) as String
                appInfo.appPackageName = pckInfo.packageName
                appInfo.appClassName = pm.getLaunchIntentForPackage(pckInfo.packageName)?.component?.className + ""
                appInfo.appCommand = randomCommand()
                data.add(appInfo)
            }
        }
        return data
    }

    private fun randomCommand(): String {
        val r = Random()
        var num = 0
        var appCommand = ""
        for (i in 0..3) {
            num = r.nextInt(10)
            appCommand += num
        }
        if (appCommand == "0000") {
            appCommand = randomCommand()
        }
        return appCommand
    }
}